django/db/models/lookups.py

```diff
@@ -190,6 +198,49 @@ class FieldGetDbPrepValueIterableMixin(FieldGetDbPrepValueMixin):
     """
     get_db_prep_lookup_value_is_iterable = True
 
+    def get_prep_lookup(self):
+        prepared_values = []
+        if hasattr(self.rhs, '_prepare'):
+            # A subquery is like an iterable, but we shouldn't independently
+            # prepare its items
+            return self.rhs._prepare(self.lhs.output_field)
+        for rhs_value in self.rhs:
+            if isinstance(rhs_value, Combinable):
+                # A combinable expression will be handled by the database, but
+                # can co-exist alongside real values
+                pass
+            elif self.prepare_rhs and hasattr(self.lhs.output_field, 'get_prep_value'):
+                rhs_value = self.lhs.output_field.get_prep_value(rhs_value)
+            prepared_values.append(rhs_value)
+        return prepared_values
+
+    def process_rhs(self, compiler, connection):
+        if self.rhs_is_direct_value():
+            # rhs should be an iterable of 2 values, we use batch_process_rhs
+            # to prepare/transform those values
+            return self.batch_process_rhs(compiler, connection)
+        else:
+            return super(FieldGetDbPrepValueIterableMixin, self).process_rhs(compiler, connection)
+
+    def resolve_combinable_parameter(self, compiler, connection, sql, param):
+        if hasattr(param, 'resolve_expression'):
+            param = param.resolve_expression(compiler.query)
+        if hasattr(param, 'as_sql'):
+            param = param.as_sql(compiler, connection)
+            param = param[0] % tuple(param[1])
+            sql = sql % param
+            param = TO_DELETE
+        return sql, param
+
+    def batch_process_rhs(self, compiler, connection, rhs=None):
+        pre_processed = super(FieldGetDbPrepValueIterableMixin, self).batch_process_rhs(compiler, connection, rhs)
+        params = []
+        for sql, param in zip(*pre_processed):
```

Comment:
I looked at this more closely, and realised I'd been a bit naïve with the implementation of resolve_combinable_parameter. I've now generalised it a bit to allow for more complex lookups and prevented a nasty sql injection vulnerability (and added a new test to cover that).